File: UsageMonitoring.Mac/Sources/UsageMonitoringMac/Models/RateLimitBucket.swift
import Foundation

struct RateLimitBucket: Codable, Identifiable, Equatable, Sendable {
    let id: String
    let label: String
    let windowDurationMins: Int?
    let usedPercent: Double
    let remainingPercent: Double
    let resetsAt: Date?
    let syncedAt: Date
    let limitId: String?
    let limitName: String?
    let windowRole: String?
}

File: UsageMonitoring.App/Models/RateLimitBucket.cs
namespace UsageMonitoring.App.Models;

public sealed record RateLimitBucket(
    string Label,
    int? WindowDurationMins,
    double UsedPercent,
    double RemainingPercent,
    DateTimeOffset? ResetsAtUtc,
    DateTimeOffset SyncedAtUtc,
    string? LimitId = null,
    string? LimitName = null,
    string? WindowRole = null);
